lib/pages/home/pins.dart

```diff
@@ -19,12 +19,107 @@ class PinsPage extends StatefulWidget {
 }
 
 class _PinsPageState extends State<PinsPage> {
+  SortType _sortType = SortType.latest;
+
+  set sortType(SortType value) {
+    _sortType = value;
+  }
+
   late final LoadingBase<PinItemModel> _lb = LoadingBase(
     request: (_, String? lastId) => RecommendAPI.getRecommendPins(
       lastId: lastId,
+      sortType: _sortType,
     ),
   );
 
+  Widget _buildPinSelectedSwitch(BuildContext context) {
+    return Container(
+      width: 90,
+      padding: const EdgeInsets.symmetric(horizontal: 4.0),
+      decoration: BoxDecoration(
+        borderRadius: RadiusConstants.r20,
+        color: Colors.grey[200],
+      ),
+      child: Stack(
+        alignment: _sortType == SortType.recommend
+            ? AlignmentDirectional.centerEnd
+            : AlignmentDirectional.centerStart,
+        children: [
+          Container(
+            margin: const EdgeInsets.all(2.0),
+            padding: const EdgeInsets.symmetric(
+              horizontal: 4.0,
+              vertical: 4.0,
+            ),
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: <Widget>[
+                GestureDetector(
+                  onTap: () => safeSetState(() {
+                    _sortType = SortType.latest;
+                    _lb.refresh(true);
+                  }),
+                  child: Text(
+                    context.l10n.pinNewest,
+                    style: context.textTheme.caption,
+                  ),
+                ),
+                GestureDetector(
+                  onTap: () => safeSetState(() {
+                    _sortType = SortType.recommend;
+                    _lb.refresh(true);
+                  }),
+                  child: Text(
+                    context.l10n.pinLatest,
+                    style: context.textTheme.caption,
+                  ),
+                ),
+              ],
+            ),
+          ),
+          Container(
+            width: _sortType == SortType.recommend ? 35 : 45,
+            height: 20,
+            decoration: const BoxDecoration(
+              borderRadius: RadiusConstants.r10,
+              color: Colors.white,
+            ),
+            child: Center(
+              child: Text(
+                _sortType == SortType.recommend ? context.l10n.pinLatest : context.l10n.pinNewest,
```

Comment:
The line seems to be overflowed with 80 indent.

Reply:
What does the meaning of overflow with 80 indent? I used Reformat code using 'dart format' and it refactors the code in this way

Reply:
Means your indent configuration is not 80 for dart files.